### PR TITLE
Hkh/add config node

### PR DESCRIPTION
**Pull request by HosseinHeris**
[] review param_config
[] review configloader node
[] review flow and see if configs are used properly 
[] run notebook as an end2end experimentation 
[] run local deploy and see if it works as expected. 
** note: I think we might need to do something similar with the output so the local deployment does not show as json file.

### CODE REVIEW

File: financial_transcripts/rag-copilot/ConfigLoader.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+
+from promptflow import tool
+import yaml
+
+# The inputs section will change based on the arguments of the tool function, after you save the code
+# Adding type to arguments and return value will help the system show the types properly
+# Please update the function name/signature per need
+@tool
+def my_python_tool(config_yaml: str) -> str:
```
Comment (journeyman-msft) [inline financial_transcripts/rag-copilot/ConfigLoader.py:9]: recommend changing default `my_python_tool` name

**Review by journeyman-msft — COMMENTED** [review]
@HosseinHeris After trying it out I discovered that
- An unintended consequence is we can no longer run each node for debugging purposes. 
image

- There are now two yaml's when you are in the notebook experience to think about. The `runs_config.yaml` and `param_config.yaml`. I see there is aa function thaat updates the `param_config.yaml` but what if the user doesn't want that - they might be confused that the last experiment is now the one for their vscode experience. 

Overall the code runs. I unfortunately think we should forego this approach until promptflow comes up with an override. What do you think?

**Comment by HosseinHeris** [comment]
@journeyman-msft : having multiple yaml files is very common in software engineering. I personally don't see any issue there. However, I did not run into the issue with the configloader. Need to double check that.

**Comment by HosseinHeris** [comment]
@journeyman-msft: I think the benefits we are getting from params.yaml approach exceeds the downsides. (1) I think it makes it easier for experimentation and deployment (2) easier for automation. The only downside is the single node running, which will likely resolve by promptflow team. Also, there is a workaround that we can always use the actual values as input for debugging purposes if needed. Another workaround is to use the debugging. I checked and it is compatible with the proposed approach.

**Review by journeyman-msft — APPROVED** [review]
Approving as functional code. I think we need to document that "node running on its own  that depend on config yaml will not work today, but should work when run overall flow"

**Comment by HosseinHeris** [comment]
@journeyman-msft: added notes to the readme.

**Pull request merged**